### PR TITLE
Have initCause() return an @Initialized throwable

### PR DESCRIPTION
**Pull request by sooniln**
The current behavior of returning an @UnknownInitialization throwable prevents common method chaining like: "new Throwable(e.getMessage().initCause(e)", as the initialization checker complains the result is not @Initialized. This exact pattern is suggested in the javadocs for this method.

In the absence of a @PolyInitialized annotation which seems like the more correct solution, I would proposed always marking the return value as @Initialized to allow such chaining.

### CODE REVIEW

**Comment by mernst** [comment]
Thanks for bringing up this example.

Unfortunately, this change is unsound, so it cannot be added to the JDK.

Here are two sound alternatives:
 * Define a `@PolyInitialized` annotation, as you suggested.
 * Write the `@Initialized` annotation on `final` subclasses of `Throwable` whose constructor does not take a cause.

> This exact pattern is suggested in the javadocs for this method.

The API documentation says the given example is for "a legacy throwable type without other support for setting the cause".
In other words, the code pattern of `new HighLevelException().initCause(le)` is **not** recommended.  Rather, it is a hack to work around user-defined exceptions that have not been updated since the `Throwable(Throwable cause)` constructor was introduced in Java 1.4 in February 2002.

**Comment by sooniln** [comment]
Well, given that many (The majority? Have not bothered to actually count...) of standard library exceptions still do not provide constructors that include a cause parameter 19 years after that constructor was introduced, I think it's safe to say the problem is not just about user code :)

Are there any plans to ever introduce @PolyInitialized?

**Comment by mernst** [comment]
Thanks for pointing out that the standard libraries are also the poor-style "legacy" code!  I didn't realize that, because the exceptions I use most do have the constructor.

The issue you have pointed out is a compelling reason to define a `@PolyInitialized` annotation.  Would you be willing to open a pull request for that?  Alternately, you could open an issue and someone will get to it when they don't have other tasks to do.

Thanks again for pointing this out.

**Comment by sooniln** [comment]
After grepping around for PolyNull, it looks like implementing PolyInitialized would be slightly non-trivial, and I have no understanding of the core internals of the checker framework. I'll open an issue for this. Thanks!
